sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -139,11 +139,13 @@ class DataFrame private[sql](
   @transient protected[sql] val logicalPlan: LogicalPlan = queryExecution.logical match {
     // For various commands (like DDL) and queries with side effects, we force query optimization to
     // happen right away to let these side effects take place eagerly.
-    case _: Command |
-         _: InsertIntoTable |
-         _: CreateTableAsSelect[_] |
-         _: CreateTableUsingAsSelect |
-         _: WriteToFile =>
+    case _ : Command =>
+      queryExecution.sparkPlan.executeCollect()
+      queryExecution.analyzed
```

Comment:
This will leads to executed command twice when we do action operator on dataframe, such as
`sql(s"CREATE DATABASE xxx").count()`
first execution is when constructing dataframe
second is to execute count.

So maybe we still need construct LocalRelation here?